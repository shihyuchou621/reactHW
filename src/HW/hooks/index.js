import HW1 from './HW1';
import HW2 from './HW2';
import HW3 from './HW3';
import HW4 from './HW4';
import HW5 from './HW5';
import HW6 from './HW6';
import HW7 from './HW7';
import HW8 from './HW8';
import HW9 from './HW9';
import HW10 from './HW10';
import HW11 from './HW11';
import HW12 from './HW12';
import HW13 from './HW13';
import HW14 from './HW14';
import HW15 from './HW15';
import HW16 from './HW16';
import HW17 from './HW17';
import HW18 from './HW18';
import HW19 from './HW19';
import HW20 from './HW20';
import HW21 from './HW21';
import HW22 from './HW22';
import HW23 from './HW23';
import HW25 from './HW25';
import HW26 from './HW26';
import HW27 from './HW27';

export default {
  HW1,
  HW2,
  HW3,
  HW4,
  HW5,
  HW6,
  HW7,
  HW8,
  HW9,
  HW10,
  HW11,
  HW12,
  HW13,
  HW14,
  HW15,
  HW16,
  HW17,
  HW18,
  HW19,
  HW20,
  HW21,
  HW22,
  HW23,
  HW25,
  HW26,
  HW27,
};