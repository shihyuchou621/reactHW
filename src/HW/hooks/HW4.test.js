import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW4';


const cmEvent = {
  target: {
    name: 'cm',
    value: '165',
  }
};

const kgEvent = {
  target: {
    name: 'kg',
    value: '50',
  }
};

const eventObj = {
  cm: cmEvent,
  kg: kgEvent,
};

const itemList = ['cm', 'kg'];

test('should HW4', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

itemList.forEach(item => {
  test(`${item} should change`, () => {
    const onChange = jest.fn(e => e.target.value);
    const { container, getByTestId } = render(
      <HW
        onChange={onChange}
      />
    );
    userEvent.type(getByTestId(item), eventObj[item].target.value);
    // console.log(eventObj[item].target.value);
    expect(container).toMatchSnapshot();
  });
});

// test('height should change', () => {
//   const DOM = mount(<HW />);
//   DOM.find('input[name="cm"]').simulate('change', cmEvent);
//   expect(DOM.debug()).toMatchSnapshot();
// })

// test('weight should change', () => {
//   const DOM = mount(<HW />);
//   DOM.find('input[name="kg"]').simulate('change', kgEvent);
//   expect(DOM.debug()).toMatchSnapshot();
// })

test('should calculate', () => {
  const onChange = jest.fn(e => e.target.value);
  const { container, getByTestId } = render(
    <HW
      onChange={onChange}
    />
  );
  const button = container.querySelector('button');
  userEvent.type(getByTestId("cm"), eventObj.cm.target.value);
  userEvent.type(getByTestId("kg"), eventObj.kg.target.value);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

