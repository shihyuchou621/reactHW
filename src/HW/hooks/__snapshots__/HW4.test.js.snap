// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cm should change 1`] = `
<div>
  <div>
    身高: 
    <input
      data-testid="cm"
      name="cm"
      value="165"
    />
  </div>
  <div>
    體重: 
    <input
      data-testid="kg"
      name="kg"
      value=""
    />
  </div>
  <button>
    計算!
  </button>
  <div>
    BMI: 
    
     
  </div>
</div>
`;

exports[`kg should change 1`] = `
<div>
  <div>
    身高: 
    <input
      data-testid="cm"
      name="cm"
      value=""
    />
  </div>
  <div>
    體重: 
    <input
      data-testid="kg"
      name="kg"
      value="50"
    />
  </div>
  <button>
    計算!
  </button>
  <div>
    BMI: 
    
     
  </div>
</div>
`;

exports[`should HW4 1`] = `
<div>
  <div>
    身高: 
    <input
      data-testid="cm"
      name="cm"
      value=""
    />
  </div>
  <div>
    體重: 
    <input
      data-testid="kg"
      name="kg"
      value=""
    />
  </div>
  <button>
    計算!
  </button>
  <div>
    BMI: 
    
     
  </div>
</div>
`;

exports[`should calculate 1`] = `
<div>
  <div>
    身高: 
    <input
      data-testid="cm"
      name="cm"
      value="165"
    />
  </div>
  <div>
    體重: 
    <input
      data-testid="kg"
      name="kg"
      value="50"
    />
  </div>
  <button>
    計算!
  </button>
  <div>
    BMI: 
    18.37
     
  </div>
</div>
`;
